python/operations.cpp

```diff
@@ -73,6 +74,56 @@ template <typename T> void bind_sort_variable(py::module &m) {
       py::call_guard<py::gil_scoped_release>());
 }
 
+void bind_experimental(py::module &m) {
+  m.def("experimental_transform",
+        [](py::object const &kernel, const Variable &a) {
+          auto fptr_address = kernel.attr("address").cast<intptr_t>();
+          auto fptr = reinterpret_cast<double (*)(double)>(fptr_address);
+          return variable::transform<double>(
+              a, overloaded{core::transform_flags::expect_no_variance_arg<0>,
+                            [&kernel](const units::Unit &u) {
+                              py::gil_scoped_acquire acquire;
+                              return py::cast<units::Unit>(kernel(u));
+                            },
+                            [fptr](const auto &x) { return fptr(x); }});
+        });
+
+  m.def("experimental_transform", [](py::object const &kernel,
+                                     const Variable &a, const Variable &b) {
+    auto fptr_address = kernel.attr("address").cast<intptr_t>();
+    auto fptr = reinterpret_cast<double (*)(double, double)>(fptr_address);
+    return variable::transform<double>(
+        a, b,
+        overloaded{
+            core::transform_flags::expect_no_variance_arg<0>,
+            core::transform_flags::expect_no_variance_arg<1>,
+            [&kernel](const units::Unit &x, const units::Unit &y) {
+              py::gil_scoped_acquire acquire;
+              return py::cast<units::Unit>(kernel(x, y));
+            },
+            [fptr](const auto &x, const auto &y) { return fptr(x, y); }});
+  });
+
+  m.def("experimental_transform", [](py::object const &kernel,
+                                     const Variable &a, const Variable &b,
+                                     const Variable &c) {
+    auto fptr_address = kernel.attr("address").cast<intptr_t>();
+    auto fptr =
+        reinterpret_cast<double (*)(double, double, double)>(fptr_address);
+    return variable::transform<double>(
+        a, b, c,
+        overloaded{core::transform_flags::expect_no_variance_arg<0>,
+                   core::transform_flags::expect_no_variance_arg<1>,
+                   core::transform_flags::expect_no_variance_arg<2>,
+                   [&kernel](const units::Unit &x, const units::Unit &y,
+                             const units::Unit &z) {
+                     py::gil_scoped_acquire acquire;
+                     return py::cast<units::Unit>(kernel(x, y, z));
+                   },
+                   [fptr](const auto &... args) { return fptr(args...); }});
+  });
+}
+
```

Comment:
We could add those operators for `Unit` (returning the same unit)?

Reply:
Oh I see. This is calling the scalar constructor. Is there a way to avoid use a different function in the kernel?

Reply:
We could of course support passing two kernels, one for units, one for data, but then it starts getting error-prone...

Maybe this is beginning to show that this is not so useful (not being able to support variances is also a big issue). Maybe other solutions should be investigated, e.g., support dask call graphs, trying to fuse operations, instead of custom transforms?